filechooser-portal/Main.vala

```diff
@@ -343,10 +335,61 @@ public class Files.FileChooserPortal : Object {
         yield;
 
         dialogs.remove (parent_window);
+        dialog.destroy ();
+
         response = _response;
         results = _results;
     }
 
+    private async bool check_overwrite_uri (string? uri) {
+        if (uri == null) {
+            return false;
+        }
+
+        var file = GLib.File.new_for_uri (uri);
+        if (file.query_exists ()) {
+            var display_name = file.get_basename ();
+            var primary = _("Replace “%s”?").printf (display_name);
+            unowned var secondary = _("Replacing this file will overwrite its current contents");
+            if (file.query_file_type (FileQueryInfoFlags.NOFOLLOW_SYMLINKS) == FileType.SYMBOLIC_LINK) {
+                try {
+                    var info = file.query_info (FileAttribute.STANDARD_SYMLINK_TARGET, FileQueryInfoFlags.NONE);
+                    if (info != null) {
```

Comment:
According to the docs, `file.query_info` doesn't return null.

Reply:
Fixed